msg.go

```diff
@@ -732,12 +732,9 @@ func (dns *Msg) PackBuffer(buf []byte) (msg []byte, err error) {
 
 	// We need the uncompressed length here, because we first pack it and then compress it.
 	msg = buf
-	compress := dns.Compress
```

Comment:
we copy the *Msg, then SetReply() to "rewrite" the response. i suppose a deep-copy would work as well (and be safer). it's a different, but related problem.

at the end of the day mutating the Compress flag and using Len() will give racy results.